src/web/extractors.rs

```diff
@@ -358,6 +408,20 @@ pub struct CollectionPostRequest {
     pub bsos: BsoBodies,
 }
 
+impl CollectionPostRequest {
+    /// Return bsos that failed to parse as appropriate failed hashmap
+    pub fn failed_bsos(&self) -> HashMap<String, String> {
+        // Leave room for any new errors when attempting to save them
+        let mut result = HashMap::with_capacity(self.bsos.invalid.len() + self.bsos.valid.len());
+        for raw in &self.bsos.invalid {
+            if let Some(id) = raw.get("id").and_then(|v| v.as_str()) {
+                result.insert(id.to_string(), "invalid json".to_string());
```

Comment:
we're cheating on the failed error messages here, python shows a more detailed message about what failed to validate

After a quick look I don't see any of our sync client code utilizing these messages, so maybe it's not too important, but I could see the messages being potentially useful for debugging (if they were echoed to a debug console log message e.g.). Very least we should log an issue about it?

Reply:
> After a quick look I don't see any of our sync client code utilizing these messages

I believe any useful error messages get converted into simple numeric codes before they reach the client anyway, so *shrug*

Reply:
> I believe any useful error messages get converted into simple numeric codes before they reach the client anyway, so _shrug_

everywhere *except* here in POST's failed mapping, the API docs even have an example of it: https://mozilla-services.readthedocs.io/en/latest/storage/apis-1.5.html#individual-collection-interaction